src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const updatedState = [];
+  let tempObject = { ...state };
+
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties':
+        add(action.extraData);
+        break;
+      case action.type === 'removeProperties':
+        remove(action.keysToRemove);
+        break;
+      case action.type === 'clear':
+        clear();
+        break;
+    }
```

Comment:
A default case should be added in the switch statement to handle any cases where action.type does not match any of the provided cases. This helps in error handling and ensures that the function behaves predictably even with unexpected inputs.